qiime2/core/cache.py

```diff
@@ -788,10 +801,24 @@ def _register_key(self, key, value, pool=False):
 
         key_fp = self.keys / key
 
+        key_dict = {}
+        key_dict['origin'] = key
+
         if pool:
-            key_fp.write_text(_KEY_TEMPLATE % (key, '', value))
+            key_dict['pool'] = value
+
+            if collection is not None:
+                key_dict['order'] = \
+                    [{k: str(v.uuid)} for k, v in collection.items()]
         else:
-            key_fp.write_text(_KEY_TEMPLATE % (key, value, ''))
+            key_dict['data'] = value
+
+            if collection is not None:
+                raise ValueError('An ordered Collection key can only be made'
+                                 ' for a pool.')
+
+        with open(key_fp, 'w') as fh:
+            yaml.safe_dump(key_dict, fh)
```

Comment:
I was thinking we may want to set `default_flow_style=False`, but after testing it with a collection, it seems to work fine. :+1: